### PR TITLE
Update functional tests to use new ragger features

### PR DESCRIPTION
**Pull request by xchapron-ledger**
@ericpassmore Can you give this PR a look? Thanks.

### CODE REVIEW

**Review by ericpassmore — COMMENTED** [review]
<del>I'll need time to do a full review. </del>. Ran tests on **physical nanosp** and found one issue. Issue linked at the bottom. All tests pass and look good on speculos emulator for nanos, nanox, and nanosp. 

Did a quick read of the code, and there are many improvements here. <del>As these are tests, and my suggestion is to merge in the changes. </del> Will continue to investigate the failure.

File: tests/functional/conftest.py [hunk 1]
```diff
@@ -85,29 +87,29 @@ def prepare_speculos_args(firmware: Firmware, display: bool):
 # Depending on the "--backend" option value, a different backend is
 # instantiated, and the tests will either run on Speculos or on a physical
 # device depending on the backend
-def create_backend(backend: str, firmware: Firmware, display: bool):
-    if backend.lower() == "ledgercomm":
+def create_backend(backend_name: str, firmware: Firmware, display: bool):
```
Comment (fbeutin-ledger) [inline tests/functional/conftest.py]: You can add the newest fixture from the ragger template

File: tests/functional/test_pubkey_cmd.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+from ragger.bip import calculate_public_key_and_chaincode, CurveChoice
+from ragger.navigator import NavInsID, NavIns
+
+from apps.eos import EosClient, ErrorType
+from utils import ROOT_SCREENSHOT_PATH
+
+# Proposed EOS derivation paths for tests ###
+EOS_PATH = "m/44'/194'/12345'"
+
+SPECULOS_EXPECTED_PUBLIC_KEY = "8e494cb179a5acef773c1bbf83ad2cd7"\
```
Comment (fbeutin-ledger) [inline tests/functional/test_pubkey_cmd.py]: remove

File: tests/functional/usage.md [hunk 3]
```diff
@@ -18,7 +18,7 @@ The application to test must be compiled for all required devices.
 You can use for this the container `ghcr.io/ledgerhq/ledger-app-builder/ledger-app-builder-lite`:
 ```
 docker pull ghcr.io/ledgerhq/ledger-app-builder/ledger-app-builder-lite:latest
```
Comment (fbeutin-ledger) [inline tests/functional/usage.md:20]: get the newest usage.md

**Review by fbeutin-ledger — DISMISSED** [review]
Looks good, a few remarks but nothing blocking
I didn't review the details too much, we'll improve as it goes

**Review by ericpassmore — DISMISSED** [review]
completed review

File: tests/functional/usage.md (hunk 3 above)
Comment (xchapron-ledger) [reply tests/functional/usage.md:20]: done

File: tests/functional/conftest.py (hunk 1 above)
Comment (xchapron-ledger) [reply tests/functional/conftest.py]: done

File: tests/functional/test_pubkey_cmd.py (hunk 2 above)
Comment (xchapron-ledger) [reply tests/functional/test_pubkey_cmd.py]: done

**Comment by xchapron-ledger** [comment]
@ericpassmore I addressed @fbeutin-ledger comments, which basically was: update to ragger 1.2.0.
So I did.
It only change a bit the expected arguments that should be pass to pytest to launch the tests: `--nanox` becomes `--device nanox` and default behavior (launching on all devices) as been replaced by an explicit `--device all`.

**Pull request merged**